src/parser/matches/arg_matches.rs

```diff
@@ -1415,6 +1612,91 @@ impl<'a> Default for GroupedValues<'a> {
     }
 }
 
+/// Macrot to reduce boilerplate for defining a new Iterator type that simply
+/// delegates to a different iterator
+macro_rules! delegating_iterator {
+    ($name:ident<$($t:tt),+> $(where Self: $lf:lifetime)?, $iter_type:ty, $item_type:ty) => {
```

Comment:
Please revert back to hand implementations.

In clap, we get built many more times than pieces like this get edited and we are trying to keep our compiler features simple.  For example, I recently dropped `?`.